File: tests/IfOr.cs
namespace Demo {
    public partial class Foo {
        // TODO: field ehVazio: bool -> declare a field
        public bool ehVazio;
        // TODO: field indUltimaPos: int -> declare a field
        public int indUltimaPos;
        public bool Check(int frowInd) {
            bool result;
            if (self.ehVazio || frowInd > self.indUltimaPos || frowInd == -1) result = true; else result = false;
            return result;
        }
    }
}

File: tests/IfOr.pas
namespace Demo;

type
  Foo = class
  private
    ehVazio: Boolean;
    indUltimaPos: Integer;
  public
    method Check(frowInd: Integer): Boolean;
  end;

implementation

method Foo.Check(frowInd: Integer): Boolean;
begin
  if ( self.ehVazio ) or
     ( frowInd > self.indUltimaPos) or
     ( frowInd = -1 )   then
    result := true
  else
    result := false;
end;

end.
